### ISSUE TITLE
UnifiedNLP not registered in Omnirom 8.1.0

### ISSUE THREAD

**Issue by mase76** [microg/android_packages_apps_UnifiedNlp#149]
Hi!
UnifiedNLP is not registered in Omnirom 8.1.0. It worked before in 7.1.2. My device is a Sony Xperia X 
What will I need to post?

**Comment by dOmuu** [comment]
I am also having this issue on a 8.1.0 based rom (FireHound on Nexus 6P)

Self-Check is all good on everything but UnifiedNLP (it's not registered in system).
I have tried installing NetworkLocation.apk manually following the instructions on the wiki - no change after a restart. 

I've since re-flashed as gsmcore has UnifiedNLP already, and having both will probably cause more headaches.

Was working no problem on Nougat as well, so might be an issue with 8.1.0

**Comment by ArchangeGabriel** [comment]
Then it’s likely that the issue is because you did not push `GmsCore.apk` to `/system/priv-app/`.

**Comment by mase76** [comment]
Yes, pushing GmsCore.apk to /system/priv-app/ did the job. I saw, there is an official patch,
which hasn't been applied yet. Why not? I don't like to flash an unofficial zip.

**Comment by ArchangeGabriel** [comment]
Can you link to the patch?

**Comment by mase76** [comment]
The patch is on the official wikipage. Scroll down:
https://github.com/microg/android_packages_apps_GmsCore/wiki/Installation

**Comment by ArchangeGabriel** [comment]
Sorry I wasn’t clear, I know about this one, I was asking whether you’ve seen somewhere it was discussed for inclusion in LineageOS4μG or OmniROM.

**Comment by mase76** [comment]
My mistake! I did not notice, that this patch has to be applied to the ROM. I thought, it has
to be applied to the GmsCore.

**Comment by bam80** [comment]
>I was asking whether you’ve seen somewhere it was discussed for inclusion in LineageOS4μG or OmniROM

@ArchangeGabriel LineageOS for microG already ships GmsCore.apk in /system, so the patch is not needed there.
However, it doesn't do the trick for [CarbonROM](https://carbonrom.org/) (Oreo): moving GmsCore to /systems seem gets Location working, but UnifiedNLP ticks in microG Self-Check still don't appear(it was rebooted).
What else could I try?

**Comment by bam80** [comment]
>I was asking whether you’ve seen somewhere it was discussed for inclusion in LineageOS4μG or OmniROM

@ArchangeGabriel LineageOS for microG already ships GmsCore.apk in /system, so the patch is not needed there.
However, it doesn't do the trick for [CarbonROM](https://carbonrom.org/) (Oreo): moving GmsCore to /systems seem gets Location working, but UnifiedNLP ticks in microG Self-Check still don't appear(it was rebooted).
What else could I try?

**Comment by mase76** [comment]
Did you also set the permission of the moved apk to 644?

**Comment by bam80** [comment]
>Did you also set the permission of the moved apk to 644?

Yes, they already were so

**Comment by realmain** [comment]
Using [LineageOS microG fork](https://lineage.microg.org/) 15.1, for the OnePlus 3T. It's causing my GPS to take forever to get a location, unlike in 14.1 when UnifiedNlp is able to bind. I have **Mozilla Location Service**, **Deja Vu Location Service**, and **Apple Wi-Fi**. as my backends. The same as when I used 14.1

![UnifiedNlp](https://i.imgur.com/xCqF1jX.png)

**Comment by utack** [comment]
Hi,  
I recently updated [my ROM ](https://forum.xda-developers.com/moto-z-play/development/rom-android-source-project-8-0-t3709130) and somehow UnifiedNlp will not register any more.
I installed the app, and moved it to /system/priv-app
Can it be because I removed com.qualcomm.location?
![screenshot_20180322-224313](https://user-images.githubusercontent.com/8551234/37800333-c312fa7c-2e22-11e8-9ed4-32a48ba10d7e.png)
Help is very welcome, thank you!

**Comment by haarp** [comment]
Same problem here. On Android 8, UnifiedNLP will not register, regardless of wether it's installed as a regular app or in /system/priv-app/.

I can instead use the [full GmsCore](https://github.com/microg/android_packages_apps_GmsCore) and put it in /system/priv-app/ to make locations work. But I don't want or need the full package, just NLP.

**Comment by Jaxseven** [comment]
I using OmniRom 9 with MicroG via NanoDroid and I've been able to get everything in Self Check except the "System supports location provider" remains unchecked. I have been trying figure out the patch linked in the Wiki but I'm not sure what to do. I've also tried adb pushing the GmsCore in the /system/pri-app/ but it doesn't seem to work for me. Are there more detailed instructions somewhere?

**Comment by mase76** [comment]
The patch is already included in Omnirom.

**Comment by Jaxseven** [comment]
So should I just disregard the unchecked setting? Will UnifiedNLP function normally regardless?

**Comment by mase76** [comment]
Do you use UnifiedNlp from Fdroid? This is a very old version and does not work. Install the Service core from the MicroG site or from their Fdroid repo.

**Comment by haarp** [comment]
@mase76 The microG Fdroid repo is also quite outdated. https://microg.org/fdroid.html

I haven't yet found a version of UNLP that works on my LineageOS.

**Comment by IzzySoft** [comment]
@haarp maybe try the Nanodroid repo (unofficial builds): https://nanolx.org/fdroid/repo/

**Comment by Jaxseven** [comment]
@IzzySoft Would the latest stable zip from Nandroid be the same as what's in the repo? I'm flashing the MicroG, F-Droid, and Webview zips.

**Comment by IzzySoft** [comment]
I don't know, I'm not involved with the Nanodroid (note additional "o" :wink:) project. I just know it also ships microG. I myself use "LineageOS for microG", so for me it already ships with the ROM.
